core/domain/question_fetchers.py

```diff
@@ -31,29 +31,26 @@
 
 
 def get_questions_and_skill_descriptions_by_skill_ids(
-        question_count, skill_ids, start_cursor):
+        question_count, skill_ids, offset=0):
     """Returns the questions linked to the given skill ids.
 
     Args:
         question_count: int. The number of questions to return.
         skill_ids: list(str). The ID of the skills to which the questions are
             linked.
-        start_cursor: str. The starting point from which the batch of
-            questions are to be returned. This value should be urlsafe.
+        offset: int. Number of query results to skip.
 
     Returns:
-        list(Question), list(list(str)), str. The list of questions and the
+        list(Question), list(list(str)), int. The list of questions and the
         corresponding linked skill descriptions which are linked to the
-        given skill ids and the next cursor value to be used for the next
-        batch of questions (or None if no more pages are left). The returned
-        next cursor value is urlsafe.
+        given skill ids and the next offset value for next batch of questions.
```

Comment:
Here and elsewhere, you should explain the possibility of None as the third return value, and update the return type accordingly.

Having said that, I think a boolean `more` value would be more appropriate here? The caller can easily calculate the next offset if `more` == True. And that way, you don't have this weird "number or None" thing.

Ditto elsewhere.

Reply:
None will be only in question_list.py. We are converting next_offset to None only in the controller question_list.py

Reply:
My suggestion still stands, since we are trying to define a general API with as few special cases as possible. I think just the info about whether there are more questions to fetch or not is needed, right?

Reply:
The main problem is that the whole frontend works on the basis of cursors. Cursors are string except when they are at the last page. So to make our approach replace the cursor approach, this step is undertaken. This information is needed because our frontend works assuming offset(earlier cursor) will be null when there are no more models to fetch.

Reply:
If the backend is now using offsets instead of cursors, the frontend needs to work in terms of offsets too. Please don't have a hybrid approach where one side is using offsets and the other is using cursors.

With offsets, you only really need a boolean (conceptually) and it should be possible to update the frontend to respect that contract. If you keep bits of the old system in then everything gets muddy.

Reply:
So do you want the backend to return the offset as an integer and introduce a new variable `more` which does the work of checking whether there are more pages?

Reply:
No -- as I have explained above, you do not need to return the new offset. Replace the third part of the return value with a boolean. The client can calculate a new offset if it needs to and if `more` is True.